enhancements/installer/single-node-installation-bootstrap-in-place.md

```diff
@@ -0,0 +1,256 @@
+3. Write RHCOS image and the master ignition to disk.
+4. Reboot the node.
+
+#### Cluster-bootstrap
+
+By default, `cluster-bootstrap` start the bootstrap control plane and create all the manifests under /opt/openshift/manifests.
+`cluster-bootstrap` also wait for a list required pods to be ready, these pods are expetced to start running on the master nodes.
+In case we are running the bootstrap in place, there is no master node that can run the pods. cluster-bootstrap should apply the manifest and tear down the control plane, in case it fail to apply some of the manifest it should return an error.
+
+
+`cluster-bootstrap` will have a new entrypoint `iBIP` which will get the master ignition as input and will enrich the master ignition with control plane static pods manifests and all required resources including etcd data.
+
+#### Bootstrap / Control plane static pods
+The control plane components we will add to the master ignition are (to be placed under `/etc/kubernetes/manifests`):
+1. etcd-pod
+2. kube-apiserver-pod
+3. kube-controller-manager-pod
+4. kube-scheduler-pod
+
+Control plane required resources to be added to the ignition:
+1. `/var/lib/etcd`
+2. `/etc/kubernetes/bootstrap-configs`
+3. /opt/openshift/tls/* (`/etc/kubernetes/bootstrap-secrets`)
+4. /opt/openshift/auth/kubeconfig-loopback (`/etc/kubernetes/bootstrap-secrets/kubeconfig`)
+
+Note: `/etc/kubernetes/bootstrap-secrets` and `/etc/kubernetes/bootstrap-configs` will be deleted post reboot once the OCP control plane is ready.
+
+The control plane opertors (that will run on the node post reboot) will manage the rollout of new revision of the control plane pods
+
+#### Etcd data
+
+In order to add a viable, working etcd post reboot.
+We will take a snapshot of etcd and add it to the master ignition.
+Oost reboot we will use restore etcd-member from the snapshot
+
+Another option is to stop the etcd pod (move the static pod manifest from `/etc/kubernetes/manifests`).
+Let etcd save it's state and exit, once etcd is down we will add the `/var/lib/etcd` directory to the master ignition.
+After the reboot etcd should start with all the data it had prior to the reboot.
+
+#### Post reboot
+
+We will add a new post-reboot service for approving the kubelet and the node Certificate Sign Requests.
+This service will also cleanup the bootstrap static pods resources once the OCP control plane is ready.
+
+### Risks and Mitigations
+
+*What are the risks of this proposal and how do we mitigate. Think broadly. For
+example, consider both security and how this will impact the larger OKD
+ecosystem.*
+
+*How will security be reviewed and by whom? How will UX be reviewed and by whom?*
+
+## Design Details
+
+### Open Questions
+
+1. What platform should it support? (just start with none) 
+2. Bootable installation artifact?, perhpas the installer should produce the liveCD?
+3. How will bootkube.sh know disk should to use for the installation? 
```

Comment:
"disk should to use" -> "which disk to use", and Colin hints at the answer [here][1].

[1]: https://github.com/openshift/enhancements/pull/565#discussion_r542456165